### PR TITLE
enhance: enhance apis by decorator and serializer class

### PR DESCRIPTION
**Pull request by W-Shih**
- enhance: enhance APIs by decorator
- enhance: enhance APIs to use its own specific serializer class
- enhance: change account API from ViewSet to GenericViewSet
- enhance CommentSerializerForCreate() by getting user id from request context
- test: react to enhancement by decorator
- fix: fix a bug in required_params() decorator where no params required

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,4 +19,4 @@ $HISTORY$
 [![codecov](https://codecov.io/gh/W-Shih/django-twitter/branch/main/graph/badge.svg?token=D5GVH7WM85)](https://codecov.io/gh/W-Shih/django-twitter)
 [![Codacy Badge](https://app.codacy.com/project/badge/Grade/dcf1d5f1fffa46ab86d5ec044d8ce7e5)](https://www.codacy.com/gh/W-Shih/django-twitter/dashboard?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=W-Shih/django-twitter&amp;utm_campaign=Badge_Grade)
 [![Codacy Badge](https://app.codacy.com/project/badge/Coverage/dcf1d5f1fffa46ab86d5ec044d8ce7e5)](https://www.codacy.com/gh/W-Shih/django-twitter/dashboard?utm_source=github.com&utm_medium=referral&utm_content=W-Shih/django-twitter&utm_campaign=Badge_Coverage)
-[![pylint Score](https://mperlet.github.io/pybadge/badges/9.64.svg)](https://app.travis-ci.com/W-Shih/django-twitter) <!-- https://mperlet.github.io/pybadge/ -->
+[![pylint Score](https://mperlet.github.io/pybadge/badges/9.63.svg)](https://app.travis-ci.com/W-Shih/django-twitter) <!-- https://mperlet.github.io/pybadge/ -->
```
Comment (wen-wayne-shih) [inline README.md:22]: Seems the pylint score is 9.64

File: accounts/api/views.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
 # 21-Aug-2021  Wayne Shih              Use rest_framework.status instead and add some comments
 # 21-Aug-2021  Wayne Shih              Add ip information in login_status for django-debug-toolbar
 # 10-Oct-2021  Wayne Shih              React to pylint checks
+# 27-Feb-2021  Wayne Shih              Enhance accout api by decorator and GenericViewSet
```
Comment (wen-wayne-shih) [inline accounts/api/views.py:17]: check spelling

File: utils/decorators.py [hunk 3]
```diff
@@ -7,6 +7,7 @@
 # =================================================================================================
 #    Date      Name                    Description of Change
 # 27-Nov-2021  Wayne Shih              Initial create
+# 27-Feb-2022  Wayne Shih              Fix a bug where no params required
```
Comment (wen-wayne-shih) [inline utils/decorators.py:10]: to specify the decorator

**Review by wen-wayne-shih — CHANGES_REQUESTED** [review]
feedback must be addressed before merging

File: README.md (hunk 1 above)
Comment (W-Shih) [reply README.md]: Resolved the suggestion

File: accounts/api/views.py (hunk 2 above)
Comment (W-Shih) [reply accounts/api/views.py]: Resolved the suggestion

File: utils/decorators.py (hunk 3 above)
Comment (W-Shih) [reply utils/decorators.py]: Resolved the suggestion

**Review by wen-wayne-shih — APPROVED** [review]
Squash commits before merging

**Pull request merged**